src/uiprotect/utils.py

```diff
@@ -360,12 +360,61 @@ def convert_video_modes(items: Iterable[str]) -> list[VideoMode]:
 
 
 def ip_from_host(host: str) -> IPv4Address | IPv6Address:
+    """
+    Resolve hostname to IP address. Supports both IPv4 and IPv6.
+
+    Args:
+        host: IP address string or hostname to resolve
+
+    Returns:
+        IPv4Address or IPv6Address object
+
+    Raises:
+        ValueError: If host cannot be resolved to an IP address
+
+    """
     try:
         return ip_address(host)
     except ValueError:
         pass
 
-    return ip_address(socket.gethostbyname(host))
+    try:
+        addr_info = socket.getaddrinfo(host, None, socket.AF_UNSPEC, socket.SOCK_STREAM)
```

Comment:
This is a blocking call. It needs to use the `await loop.getaddrinfo` instead or aiodns

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.getaddrinfo